app/services/plugin_service.py

```
@@ -0,0 +1,61 @@
+class PluginService:
+    @staticmethod
+    async def create_plugin_config(user_id: str, config: PluginConfigurationCreate) -> dict:
+        # Encrypt sensitive config using KeyService (placeholder)
+        encrypted_blob = KeyService.encrypt_config(config.encrypted_config_blob, user_id)
+        data = config.dict()
+        data["user_id"] = user_id
+        data["encrypted_config_blob"] = encrypted_blob
+        res = supabase.table(PLUGIN_TABLE).insert(data).execute()
+        if res.data:
+            return res.data[0]
+        raise Exception(res.error)
```

Ruff (0.8.2)

Create your own exception

(TRY002)